Added env variable for namespace. Moved from openshift-client to kubernetes Python package.

EldritchJS opened the pull request:
This adds the COLDPRESS_NAMESPACE env variable. The move to kubernetes python package defaults to, well, default for its namespace which per issue #18 makes things unhappy. To test I set COLDPRESS_NAMESPACE to coldpress. Also made a namespace with `oc project coldpress` prior! 

Running the basic_test example passes when the above is in place. 

This PR also eliminates the use of os.system for copying files from the pods.

larsks reviewed (commented):
This isn't a full review of this PR, but there are two requests I would like to make:

1. Try to keep pull requests focused on a single feature, like "use Python kubernetes client instead of openshift-client". In this PR, I see:

    - The switch to using the `kubernetes` module.
    - Changes to how data is copied from pods (maybe related to the previous point, but you called this change out explicitly in the PR description)
    - The addition of `streamlit` as a dependency, which seems unrelated to anything else.
    - Changes to support the `COLDPRESS_NAMESPACE` variable

    Keeping unrelated changes in separate pull requests means that small changes get merged faster, rather than being held up while people review (and possibly request changes for) more substantive changes. Having a solid focus to the PR also helps the reviewer keep the appropriate context in their head when reviewing the code.

    I would at least remove `streamlit` from this PR; it's possible that the other changes are tied to the new `kubernetes` based code such that it doesn't make sense to split them out into separate PRs; I leave that to your discretion.

2. Be definite about your commit messages. When I see this:

    > This fix seems to handle both...

    Or this:

    > This appears to permit

    I get nervous. Tell me what the commit does. If you're not sure, take the time to become sure. A common guideline for writing commit messages is to write them in the active voice, and assume that they are prefixed by the phrase "This commit will...". That would get you, for example:

    > Introduce more robust handling of COLDPRESS_NAMESPACE env variable

    Or:

    > Permit using the default namespace

EldritchJS commented:
@larsks Thank you for the insights. I added the env var in the initial commit as the need for it was made manifest after moving to the Python kubernetes package. I have removed the other unrelated commits and shall put them in their own feature branches and PRs. I have also removed the unneeded packages from requirements.txt.

larsks commented on coldpress.py:
```diff
@@ -57,26 +57,32 @@ def __init__(self):
+        try:
+            config.load_incluster_config()
+        except:
+            config.load_kube_config()
```
This is probably something for a future PR, but I would move the kubernetes initialization out of this class. There are no tests now, but at some point (e.g., yesterday) we're going to want tests, and the ability to test things out without implicitly initializing the kubernetes connection (e.g., because we're providing a fake client for testing) will be valuable.

larsks commented on openshift_runtime.py:
```diff
@@ -2,9 +2,76 @@
+        stderr_output = tar_process.stderr.read().decode()
+        raise Exception(f"tar extraction failed: {stderr_output}")
+
+    return tar_process.returncode
```
There's no reason to return a value here: it's always going to be 0 (because if `tar_process.returncode` is not zero, we throw an exception).

EldritchJS replied:
Good call, will change this.

larsks commented on openshift_runtime.py:
```diff
@@ -2,9 +2,76 @@
+v1 = client.CoreV1Api()
+
+# Get namespace from environment variable, default to "default"
+NAMESPACE = os.getenv("COLDPRESS_NAMESPACE", "default")
```
The [principal of least surprise](https://en.wikipedia.org/wiki/Principle_of_least_astonishment) suggests that we may want to default to "the current namespace", as defined by the local `kubeconfig` file when running locally or by the service account information when running in a cluster.

The best time to figure that out is when loading the config, since that's when you know whether you're running in or out of a cluster. Something like:

```python
# Initialize Kubernetes client
try:
    config.load_incluster_config()
    current_namespace = get_namespace_from_serviceAcount()
except:
    config.load_kube_config()
	current_namespace = get_namespace_from_kubeconfig()

NAMESPACE = os.getenv('COLDPRESS_NAMESPACE', current_namespace or 'default')
```

Where the named functions might look like:

```python
  def get_current_namespace_from_serviceAccount() -> str:
      namespace_path: Path = Path(
          "/var/run/secrets/kubernetes.io/serviceaccount/namespace"
      )

      if namespace_path.is_file():
          with namespace_path.open("r") as f:
              return f.read().strip()

      # Fallback if the file isn't present (e.g., service account not mounted)
      return "default"

  def get_current_namespace_from_kubeconfig() -> str:
      try:
          contexts, active_context = config.list_kube_config_contexts()
      except config.ConfigException:
          return "default"

      namespace = active_context["context"].get("namespace")

      if not namespace:
          namespace = "default"

      return namespace
```

EldritchJS commented on openshift_runtime.py:
```diff
@@ -218,28 +282,31 @@ def openshift_run(params):
+                        if e.status == 404:
+                            pass  # Pod not yet created, wait
+                        else:
+                            raise
```
👍
